src/main/java/core/basesyntax/utils/generate/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.utils.generate.impl;
+
+import core.basesyntax.storage.impl.StorageImpl;
+import core.basesyntax.utils.generate.ReportGenerator;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private final ReportGenerator generator;
+
+    public ReportGeneratorImpl(ReportGenerator generator) {
+        this.generator = generator;
+    }
+
+    @Override
+    public String getReport(StorageImpl storage) {
+        return generator.getReport(storage);
```

Comment:
Directly using `StorageImpl` as a parameter type violates the Dependency Inversion Principle, one of the SOLID principles. Consider using an interface to depend upon abstractions rather than concrete implementations.